### DISCUSSION TITLE
@types/history

### DISCUSSION THREAD

**Discussion by semtselishev** [DefinitelyTyped/DefinitelyTyped#58209] [Issues with a @types package] [answered]
`createMemoryHistory` from `@types/history@4.9.0` has incorrect types for  `initialEntries` prop. 
It can accept not only `string`:
```
export interface MemoryHistoryBuildOptions {
    getUserConfirmation?: typeof getConfirmation | undefined;
    initialEntries?: string[] | undefined;
    initialIndex?: number | undefined;
    keyLength?: number | undefined;
}
```

but also an object with `state`, `search` ... in it

![image](https://user-images.githubusercontent.com/89021574/149635559-a89334a7-8330-42b3-88ff-63db1fbb8d68.png)

because of that, this code is not compiled in TS:
```
createMemoryHistory({
    initialEntries: [{
      pathname: initialPath,
      search: initialSearch,
      initialState: initialState
    } ],
  });
```

[a comment by an automated account is not shown]

**Comment by martin-badin** [comment]
Hi, @semtselishev. I create PR to fix this issue #58217

**Comment by martin-badin** [comment]
@semtselishev Could you use the newest version of the package? The newest version of the package provides its own types.

**Reply by semtselishev** [reply]
Sure, I will try to update to history@5.x.x

**Reply by martin-badin** [reply]
PR was merged too if you still want to use the older version. :)
